src/main/java/io/github/janmalch/kino/control/reservation/GetAllReservationsControl.java

```diff
@@ -0,0 +1,52 @@
+package io.github.janmalch.kino.control.reservation;
+
+import io.github.janmalch.kino.api.model.ReservationInfoDto;
+import io.github.janmalch.kino.api.model.SeatForReservationDto;
+import io.github.janmalch.kino.control.Control;
+import io.github.janmalch.kino.control.ResultBuilder;
+import io.github.janmalch.kino.entity.Reservation;
+import io.github.janmalch.kino.entity.Seat;
+import io.github.janmalch.kino.repository.Repository;
+import io.github.janmalch.kino.repository.RepositoryFactory;
+import io.github.janmalch.kino.util.ReflectionMapper;
+import io.github.janmalch.kino.util.either.EitherResultBuilder;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GetAllReservationsControl implements Control<List<ReservationInfoDto>> {
+
+  private final ReflectionMapper<Seat, SeatForReservationDto> seatMapper = new ReflectionMapper<>();
+  private final ReflectionMapper<Reservation, ReservationInfoDto> reservationMapper =
+      new ReflectionMapper<>();
+  private final Repository<Reservation> reservationRepository =
+      RepositoryFactory.createRepository(Reservation.class);
+
+  @Override
+  public <T> T execute(ResultBuilder<T, List<ReservationInfoDto>> result) {
+    var reservations = reservationRepository.findAll();
+    var reservationDtos =
+        reservations
+            .stream()
+            .map(
+                r -> {
+                  var control = new GetReservationByIdControl(r.getId());
```

Comment:
Könnte hier der `GetReservationMapper` genutzt werden?

Reply:
Using Mapper instead of Control

https://github.com/JanMalch/kino/blob/15926a15881e5d23ff95161a94a56610538cc51b/src/main/java/io/github/janmalch/kino/control/reservation/GetMyReservationsControl.java#L23-L31